src/test/java/net/technolords/micro/input/ConfigurationSelectorTest.java

```diff
@@ -0,0 +1,114 @@
+        for(String key : testConfigs.keySet()) {
+            Configuration actualConfiguration = configurationSelector.findMatchingConfiguration(url, testConfigs);
+            LOGGER.info("Equals: {}", actualConfiguration.equals(expectedConfiguration));
+            Assert.assertEquals(actualConfiguration,expectedConfiguration);
+        }
+    }
+
+    private Map<String,Configuration> getConfigs() {
+        Map<String, Configuration> getConfigurations = new HashMap<>();
+        getConfigurations.put("/mock/get",getConfig1());
+        getConfigurations.put("/mock/*/get",wildCardConfig());
+        getConfigurations.put("/mock/1/get",getConfig2());
+        getConfigurations.put("/mock/*/get/*/data",wildCardConfig2());
+        return getConfigurations;
+    }
+
+    private Map<String, Configuration> postConfigs() {
+        Map<String, Configuration> postConfigurations = new HashMap<>();
+        postConfigurations.put("/mock/post",postConfig1());
+//        postConfigurations.put(null,emptyConfig());
+        return postConfigurations;
+    }
+
+    private Configuration getConfig1(){
+        Configuration configuration1 = new Configuration();
+        configuration1.setUrl("/mock/get");
+        configuration1.setType("GET");
+        simpleResource.setResource("configManagerTest/getResponse1.txt");
+        configuration1.setSimpleResource(simpleResource);
+        return configuration1;
+    }
+
+    private Configuration wildCardConfig(){
+        Configuration configuration2 = new Configuration();
+        configuration2.setUrl("/mock/*/get");
+        configuration2.setType("GET");
+        simpleResource.setResource("configManagerTest/getResponse2.txt");
+        configuration2.setSimpleResource(simpleResource);
+        return configuration2;
+    }
+
+    private Configuration getConfig2(){
+        Configuration configuration3 = new Configuration();
+        configuration3.setUrl("/mock/1/get");
+        configuration3.setType("GET");
+        simpleResource.setResource("configManagerTest/getResponse1.txt");
+        configuration3.setSimpleResource(simpleResource);
+        return configuration3;
+    }
+
+    private Configuration wildCardConfig2(){
+        Configuration configuration4 = new Configuration();
+        configuration4.setUrl("/mock/*/get/*/data");
+        configuration4.setType("GET");
+        simpleResource.setResource("mock/sample-get.json");
+        configuration4.setSimpleResource(simpleResource);
+        return configuration4;
+    }
+
+    private Configuration postConfig1(){
+        Configuration configuration5 = new Configuration();
+        configuration5.setUrl("/mock/*/get/*/data");
+        configuration5.setType("POST");
+        simpleResource.setResource("mock/sample-post4.txt");
+        configuration5.setSimpleResource(simpleResource);
+        return configuration5;
+    }
+
+    private Configuration emptyConfig(){
```

Comment:
You don't need to configure an empty instance.... the logic of the class to test will simply return a null if there is no match. That is different than returning an instance (implies a match) with nothing set....